### ISSUE TITLE
Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path

### ISSUE THREAD

**Issue by carlosjgp** [argoproj/argo-cd#2933]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**
Everything looks ok on the UI and my apps are in sync but the debug logs are reporting

```
time="2019-12-31T09:09:46Z" level=debug msg="Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path: \"/spec/template/spec/securityContext/runAsGroup\""
time="2019-12-31T09:09:46Z" level=debug msg="Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path: \"/spec/template/spec/containers/0/securityContext/runAsGroup\""
time="2019-12-31T09:09:46Z" level=debug msg="Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path: \"/spec/template/spec/containers/1/securityContext/runAsGroup\""
time="2019-12-31T09:09:46Z" level=debug msg="Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path: \"/spec/template/spec/containers/2/securityContext/runAsGroup\""
time="2019-12-31T09:09:46Z" level=debug msg="Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path: \"/spec/template/spec/containers/3/securityContext/runAsGroup\""
```

Notice the extra `\"` at the end of the paths... looks like it's the same as this one https://github.com/kubernetes/kubernetes/issues/63247 

**To Reproduce**

Add this block to the global configuration

```
apiVersion: v1
kind: ConfigMap
metadata:
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
  name: argocd-cm
  namespace: argocd
data:
  resource.customizations: |
    apps/Deployment:
      ignoreDifferences: |
        jsonPointers:
        - /spec/template/spec/securityContext/runAsGroup
        - /spec/template/spec/containers/0/securityContext/runAsGroup
        - /spec/template/spec/containers/1/securityContext/runAsGroup
        - /spec/template/spec/containers/2/securityContext/runAsGroup
        - /spec/template/spec/containers/3/securityContext/runAsGroup
```

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

**Expected behavior**

I'm not experiencing any noticeable issues so... everything as it is?

**Screenshots**

**Version**

ArgoCD 1.3.6 HA deployment

```shell
argocd: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:46:45Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:47:48Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: Version: {Version:kustomize/v3.2.1 GitCommit:d89b448c745937f0cf1936162f26a5aac688f840 BuildDate:2019-09-27T00:10:52Z GoOs:linux GoArch:amd64}
  Helm Version: v2.15.2
  Kubectl Version: v1.14.0
```

**Logs**

They are above :point_up: on the description

**Comment by carlosjgp** [comment]
> Notice the extra `\"` at the end of the paths... looks like it's the same as this one kubernetes/kubernetes#63247
I just noticed that the extra `\"` is not relevant since is scaping the log line... so it should be something else

**Comment by alexmt** [comment]
Hello @carlosjgp ,

These logs are printed when the resource does not have fields specified by the json pointer. It does not really indicate any error, just debugging information that is useful for troubleshooting.

**Comment by carlosjgp** [comment]
Thanks for the feedback @alexmt .

Personally I found it confusing since I have more than a handled ArgoCD applications few of them were failing and I enabled the debug level to find out why there was problems with synchronisation.

Does anyone knows if ArgoCD is going to use correlation IDs for logs? Per channel, app, cycle...?

The log statement as it is doesn't helps to find out the problem
`time="2019-12-31T09:09:46Z" level=debug msg="Failed to apply normalization: jsonpatch remove operation does not apply: doc is missing path: \"/spec/template/spec/containers/3/securityContext/runAsGroup\""`

I think I can close the issue.
